assets/js/phoenix.js

```diff
@@ -859,14 +859,15 @@ export class Socket {
    * @example socket.onOpen(function(){ console.info("the socket was opened") })
    *
    * @param {Function} callback
+   * @param {Function} callback
    */
-  onOpen(callback){ this.stateChangeCallbacks.open.push(callback) }
+  onOpen(callback, ref = this.ref){ this.stateChangeCallbacks.open.push({ [ref]: callback }) }
```

Comment:
This default would be for non internal calls to this method since it isn't a required API argument

Reply:
let's kill the ref argument and always generate one with `this.makeRef()`, then well return it to the caller to optionally use for `off/1`